op.c

```diff
@@ -15679,7 +15692,19 @@ Perl_coresub_op(pTHX_ SV * const coreargssv, const int code,
             }
             return o;
         default:
-            o = op_convert_list(opnum,OPf_SPECIAL*(opnum == OP_GLOB),argop);
+            /* For open(), OPf_SPECIAL indicates we saw a literal undef as the
+             * filename argument and thus a &PL_sv_undef argument at runtime
+             * should trigger the creation of a temp file. This is to
+             * distinguish between open(..., ..., undef) and
+             * open(..., ..., delete $hash{key}), which also passes
```

Comment:
The comments for `OPf_SPECIAL` in op.h needs to be updated to describe the usage for OP_OPEN.

Reply:
Thanks; done.